src/main/cpp/rollingfileappender.cpp

```diff
@@ -308,7 +308,9 @@ bool RollingFileAppenderSkeleton::rolloverInternal(Pool& p)
 								catch (std::exception& ex)
 								{
 									LogLog::warn(LOG4CXX_STR("Exception on rollover"));
-									errorHandler->error(ex.what(), ex, 0);
+									LogString exmsg;
+									log4cxx::helpers::Transcoder::decode(ex.what(), exmsg);
+									errorHandler->error(exmsg, ex, 0);
```

Comment:
Another good example for maybe letting `decode` return a value in future, like [discussed](https://github.com/apache/logging-log4cxx/pull/60#issuecomment-814452988) already. Besides that, using the DECODE-macro might be easier here:

    LOG4CXX_DECODE_CHAR(exmsg, ex.what());